requestParser/requestParser_test.go

```diff
@@ -249,6 +252,114 @@ func TestParseRequest(t *testing.T) {
 	}
 }
 
+func TestEventDetailsMetrics(t *testing.T) {
```

Comment:
It's a little unclear to me what the goal of this test is.  There's a `parseRequest()` test already where you can add verifying that the metric is added to when we expect it.  It seems to me like the rest of this test is covered with `TestGetEventDestinationType()`?  It seems like you're doing a lot of setup work for unrelated things in your test, which is what makes me question why this is needed.

Reply:
The goal of this test is to make sure that the events metric is added to properly. @kcajmagic suggested that we have a function just for testing the metrics, since the number of test cases were getting high. I wanted to make sure that all the different event types were added to the metric properly, but I think you're right in pointing out that `TestGetEventDestinationType` covers that, so perhaps we can just have two test cases: one for a proper event destination and existent partner ID array and the other where there are no partner IDs, and then just merge this function with `TestParseRequest`

Reply:
yeah, `TestParseRequest` is really long...I think we should refactor parseRequest() to split it up more and make the unit test simpler, but you don't need to do any of that for this pr. 🙂 In the future, hopefully we'll make it nicer.